### PR TITLE
Add GitHub action to compile and run tests for the react app

### LINKED ISSUE #3: Add GitHub action to compile and run tests for the react app

### PR DESCRIPTION
**Pull request by ngkelly**
Closes #3.
A workflow has been created to run tests when pushing and creating pull
requests against the main branch.

### CODE REVIEW

**Review by the-c0d3br34k3r — CHANGES_REQUESTED** [review]
Please add the badge/s to the README.

File: .github/workflows/run_tests.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Unit Tests
```
Comment (the-c0d3br34k3r) [inline .github/workflows/run_tests.yml:4]:
```suggestion
name: npm build
```

File: .github/workflows/run_tests.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: npm ci
+    - run: npm run build --if-present
+    - run: npm test -- --coverage
```
Comment (the-c0d3br34k3r) [inline .github/workflows/run_tests.yml:34]: The "coverage" portion should be added as a separate PR modifying the same action, addressing #4.

**Review by the-c0d3br34k3r — CHANGES_REQUESTED** [review]

File: .github/workflows/run_tests.yml (hunk 1 above)
Comment (ngkelly) [reply .github/workflows/run_tests.yml]: Done.

File: .github/workflows/run_tests.yml (hunk 2 above)
Comment (ngkelly) [reply .github/workflows/run_tests.yml]: The "coverage" portion has been removed and will be included in a separate PR.

**Review by the-c0d3br34k3r — APPROVED** [review]

**Pull request merged**